daemon/api.go

```diff
@@ -585,21 +585,50 @@ func deleteOp(c *Command, r *http.Request) Response {
 	}
 }
 
+type licT struct {
+	Intro   string
+	License string
+	Agreed  bool
+}
+
+func (*licT) Error() string {
+	// licT isn't an error, strictly speaking, but can
+	// be converted into one for convenience
+	return "license agreement required"
+}
+
 type packageInstruction struct {
+	progress.NullProgress
```

Comment:
I see, thanks. So we do want to silence the progress metering aspect, but are using the progress.Meter interface to ask questions to the user. The are two things that makes this feel slightly uncomfortable:

1. We have an Agreed function on progress.Meter, requiring interaction with the user (not progress metering, really)

2. We are doing a round trip inside snappy.Install just so it calls Agreed on the progress meter, and we know from the outside whether this will go wrong or not. Perhaps we could have the license agreement logic completely out of snappy.Install.

That said, it sounds fine to not do that now.